OpenRA.Mods.Common/Warheads/DamageWarhead.cs

```diff
@@ -50,8 +50,20 @@ public override void DoImpact(in Target target, WarheadArgs args)
 				if (!IsValidAgainst(victim, firedBy))
 					return;
 
-				var closestActiveShape = victim.TraitsImplementing<HitShape>().Where(Exts.IsTraitEnabled)
-					.MinByOrDefault(t => t.DistanceFromEdge(victim, victim.CenterPosition));
+				HitShape closestActiveShape = null;
+				var closestDistance = new WDist(int.MaxValue);
+				foreach (var targetPos in victim.GetEnabledTargetablePositions())
```

Comment:
What is the problem with keeping this as LINQ statement?
```
var closestActiveShape = victim.GetEnabledTargetablePositions().MinByOrDefault(t => t.DistanceFromEdge(victim, victim.CenterPosition));
```

Reply:
Actually I think I like the LINQ version more since it is less verbose.

Reply:
I wanted to use your example. However i need both the hiy shape and distance. As the hitshape is passed below.

Reply:
That should return the closest hitshape. The distance shouldn't be needed below?

Reply:
I looked at the `SpreadDamageWarHead`. There both are used. 

I changed it for this file.